tensorflow/python/ops/array_ops.py

```diff
@@ -1296,20 +1296,20 @@ def concat(values, axis, name="concat"):
   Returns:
     A `Tensor` resulting from concatenation of the input tensors.
   """
-  if not isinstance(values, (list, tuple)):
-    values = [values]
-  # TODO(mrry): Change to return values?
-  if len(values) == 1:  # Degenerate case of one tensor.
-    # Make a throwaway call to convert_to_tensor to make sure
-    # that axis is of the correct type, and make sure that
-    # the returned tensor is a scalar.
-    # TODO(keveman): Implement a standalone type and shape checker.
-    with ops.name_scope(name) as scope:
+  with ops.name_scope(name, values=[values]):
+    if not isinstance(values, (list, tuple)):
+      values = [values]
+    # TODO(mrry): Change to return values?
+    if len(values) == 1:  # Degenerate case of one tensor.
+      # Make a throwaway call to convert_to_tensor to make sure
+      # that axis is of the correct type, and make sure that
+      # the returned tensor is a scalar.
+      # TODO(keveman): Implement a standalone type and shape checker.
       ops.convert_to_tensor(
           axis, name="concat_dim",
           dtype=dtypes.int32).get_shape().assert_has_rank(0)
-      return identity(values[0], name=scope)
```

Comment:
Wouldn't a simpler fix be just to pass name=name to identity?

Reply:
Thanks @alextp, indeed that should fix the issue. The PR has been updated.